test/sql/catalog/dependencies/test_alter_dependency_ownership.test

```diff
@@ -0,0 +1,296 @@
+# name: test/sql/catalog/dependencies/test_alter_dependency_ownership.test
+# description: Tests alter of ownership of sequences
+# group: [dependencies]
+
+# TEST: If the table is dropped, then the sequence is also droppped
+statement ok
+CREATE SEQUENCE tablename_colname_seq;
+
+statement ok
+CREATE TABLE tablename (
+    colname integer DEFAULT nextval('tablename_colname_seq')
+);
+
+statement ok
+ALTER SEQUENCE tablename_colname_seq OWNED BY tablename;
+
+statement error
+DROP TABLE tablename;
+
+statement ok
+DROP TABLE tablename CASCADE;
+
+statement error
+SELECT nextval('tablename_colname_seq');
+
+statement ok
+CREATE SEQUENCE tablename_colname_seq;
+
+statement ok
+DROP SEQUENCE tablename_colname_seq;
+
+# TEST: If the sequence is owned by the table, inserts work normally
+statement ok
+CREATE SEQUENCE tablename_colname_seq;
+
+statement ok
+CREATE TABLE tablename (
+    colname integer DEFAULT nextval('tablename_colname_seq')
+);
+
+statement ok
+INSERT INTO tablename VALUES (default);
+
+query I
+SELECT colname FROM tablename;
+----
+1
+
+statement ok
+ALTER SEQUENCE tablename_colname_seq OWNED BY tablename;
+
+statement ok
+INSERT INTO tablename VALUES (default);
+
+statement ok
+INSERT INTO tablename VALUES (default);
+
+query I
+SELECT colname FROM tablename;
+----
+1
+2
+3
+
+statement ok
+DROP TABLE tablename CASCADE;
+
+# TEST: the sequence can be droppped but then the table loses its default value
```

Comment:
I was trying to mimic Postgres behavior, afaik they allow an owned sequence to be dropped and the columns that use the sequence as default lose the default value. In any case, I am happy with what you said, since this will be used internally anyway.

Reply:
You are correct, but it requires `DROP SEQUENCE .. CASCADE`. In their case the "dependency unit" is the default value, whereas for us the dependency unit is the entire table. I think mimicking that exactly (having the dependency unit be the default value) is not required and just dropping the table is fine.